src/pkg/backup/details/details.go

```diff
@@ -470,11 +483,35 @@ func (d *Details) add(
 		entry.ItemRef = withoutMetadataSuffix(entry.ItemRef)
 	}
 
+	if d.repoRefToEntry == nil {
+		d.repoRefToEntry = make(map[string]*Entry)
+	}
+
+	d.repoRefToEntry[entry.RepoRef] = &entry
 	d.Entries = append(d.Entries, entry)
```

Comment:
I don't think this will behave the way we want it to. `repoRefToEntry` has a pointer to the location of the passed in details but `Entries` has a copy of the passed in details. Furthermore, `Entries` can be reallocated if the number of entries outgrows the capacity of the slice. That means a pointer to an entry obtained by taking the address of what's in the slice (even if it's the entry we just appended) may not remain valid outside this function

Reply:
Oh! My understanding was that the reference to Entry is appended to slice. Just confirmed thats not true.
Thanks! I'll fix this.